### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [eriknyquist/librxvm#8]
[An extra null pointer check is not needed](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html "Do I need to check for null before free p?") in functions like the following.
- [inst_stack_cleanup](https://github.com/eriknyquist/librxvm/blob/946ec4d4a91e01efede9fbcbd50e62527919572a/librxvm/stage1.c#L44)
- [rxvm_free](https://github.com/eriknyquist/librxvm/blob/c4d41fb1ffe4f8fd8f8f479eaaf7d078c2445abc/librxvm/rxvm.c#L536)
- [vm_cleanup](https://github.com/eriknyquist/librxvm/blob/c4d41fb1ffe4f8fd8f8f479eaaf7d078c2445abc/librxvm/vm.c#L202)

**Comment by eriknyquist** [comment]
Thanks for that, good point

**Comment by eriknyquist** [comment]
fixed
54e78752c84e3bb1ca0a93dcae9186d035281a2e

**Comment by elfring** [comment]
Thanks for [your small source code improvement](https://github.com/eriknyquist/librxvm/commit/54e78752c84e3bb1ca0a93dcae9186d035281a2e "Remove unnecessary null pointer checks before free()").

Were [any update candidates left over](https://github.com/eriknyquist/librxvm/blob/cae348f74edbd28879745d5e09ac2f9d0571d2ec/librxvm/rxvm.c#L561 "Update candidate: rxvm_free()") for such a change pattern?

**Comment by eriknyquist** [comment]
No, the check in rxvm_free serves another purpose besides protecting the free(). If compiled->simple is non-NULL, then I know that compiled->exe was never allocated so I don't need to traverse/free it. It needs to stay.

**Comment by elfring** [comment]
How do you think about to integrate the following small change?

```diff
         for (i = 0; i < compiled->size; i++) {
-            if (compiled->exe[i].ccs != NULL)
-                free(compiled->exe[i].ccs);
+            free(compiled->exe[i].ccs);
         }
```
